### PR TITLE
[SPARK-49958][PYTHON] Python API for string validation functions

### PR DESCRIPTION
**Pull request by uros-db**
### What changes were proposed in this pull request?
Adding the Python API for the 4 new string validation expressions:
- is_valid_utf8
- make_valid_utf8
- validate_utf8
- try_validate_utf8

### Why are the changes needed?
Offer a complete Python API for the new expressions in Spark 4.0.

### Does this PR introduce _any_ user-facing change?
Yes, adding Python API for the 4 new Spark expressions.

### How was this patch tested?
New tests for the Python API.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by MaxGekk — CHANGES_REQUESTED** [review]
Please, merge the changes into https://github.com/apache/spark/pull/48454

**Review by uros-db — COMMENTED** [review]
first time adding Scala/Python APIs, so separate PRs allow easier review, corrections and error identifications - let's proceed with Python API changes in this PR if you're on board @MaxGekk

File: python/pyspark/sql/tests/connect/test_connect_function.py [hunk 1]
```diff
@@ -2078,6 +2078,30 @@ def test_string_functions_multi_args(self):
         cdf = self.connect.sql(query)
         sdf = self.spark.sql(query)
 
+        # test is_valid_utf8
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_connect_function.py:2081]: you can just add the test at `test_functions.py`. The tests will be resused.

**Review by HyukjinKwon — COMMENTED** [review]
can you also list them in `python/docs/source/reference/pyspark.sql/functions.rst`?

File: python/pyspark/sql/tests/connect/test_connect_function.py (hunk 1 above)
Comment (uros-db) [reply python/pyspark/sql/tests/connect/test_connect_function.py]: moved tests to `test_functions.py`

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -11197,6 +11197,127 @@ def encode(col: "ColumnOrName", charset: str) -> Column:
...
+def validate_utf8(str: "ColumnOrName") -> Column:
+    """
+    Returns the input value if it corresponds to a valid UTF-8 string, or emits an error otherwise.
+
+    .. versionadded:: 4.0.0
+
+    Parameters
+    ----------
+    str : :class:`~pyspark.sql.Column` or str
+        A column of strings, each representing a URL-encoded string.
```
Comment (MaxGekk) [inline python/pyspark/sql/functions/builtin.py:11271]: URL-encoded string? really?

Comment (uros-db) [reply python/pyspark/sql/functions/builtin.py]: thanks, fixed

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @uros-db and @HyukjinKwon for review.

**Pull request closed without merging**